test/e2e/reconciler/deployment/deployment_coordinator_test.go

```diff
@@ -0,0 +1,349 @@
+		framework.Eventually(t, func() (bool, string) {
+			if err := createResources(t, ctx, workloads.FS, upstreamConfig, workspace.name, func(bs []byte) ([]byte, error) {
+				yaml := string(bs)
+				yaml = strings.Replace(yaml, "replicas: 1", fmt.Sprintf("replicas: %d", workspace.requestedReplicas), 1)
+				return []byte(yaml), nil
+			}); err == nil {
+				return true, ""
+			} else {
+				return false, err.Error()
+			}
+		}, wait.ForeverTestTimeout, time.Millisecond*100, "should create the deployment after the deployments resource is available in workspace %q", workspace.name)
+
+		t.Logf("Wait for the workload in workspace %q to be started and available with 4 replicas", workspace.name)
+		framework.Eventually(t, func() (success bool, reason string) {
+			deployment, err := upstreamKubeClusterClient.Cluster(workspace.name).AppsV1().Deployments("default").Get(ctx, "test", metav1.GetOptions{})
+			require.NoError(t, err)
+
+			// TODO(davidfestal): the 2 checks below are necessary here to avoid the test to be flaky since for now the coordination
+			// controller doesn't delay the syncing before setting the transformation annotations.
+			// So it could be synced with 8 replicas on each Synctarget at start, for a very small amount of time, which might
+			// seem like deployment replicas would have been spread, though in fact it is not.
+			if _, exists := deployment.GetAnnotations()["experimental.spec-diff.workload.kcp.dev/"+eastSyncer.ToSyncTargetKey()]; !exists {
+				return false, fmt.Sprintf("Deployment %s/%s should have been prepared for transformation by the coordinator for the east syncTarget", workspace.name, "test")
+			}
+			if _, exists := deployment.GetAnnotations()["experimental.spec-diff.workload.kcp.dev/"+westSyncer.ToSyncTargetKey()]; !exists {
+				return false, fmt.Sprintf("Deployment %s/%s should have been prepared for transformation by the coordinator for the west syncTarget", workspace.name, "test")
+			}
+
+			// TODO(davidfestal): the 2 checks below are necessary here to avoid the test to be flaky since for now the coordination
+			// controller doesn't delay the syncing before setting the transformation annotations.
+			// So it could be synced with 8 replicas on each Synctarget at start, for a very small amount of time, which might
+			// seem like deployment replicas would have been spread, though in fact it is not.
+			if _, exists := deployment.GetAnnotations()["diff.syncer.internal.kcp.dev/"+eastSyncer.ToSyncTargetKey()]; !exists {
+				return false, fmt.Sprintf("Status of deployment %s/%s  should have been updated by the east syncer", workspace.name, "test")
+			}
+			if _, exists := deployment.GetAnnotations()["experimental.spec-diff.workload.kcp.dev/"+westSyncer.ToSyncTargetKey()]; !exists {
+				return false, fmt.Sprintf("Status of deployment %s/%s  should have been updated by the west syncer", workspace.name, "test")
+			}
+
+			dumpEventsAndPods(wkspDownstreamInfo)
```

Comment:
You're grabbing events and pod logs twice per second? Do we need this all the time, or only at the end if the test fails?